packages/ui/src/machines/authenticator/defaultServices.ts

```diff
@@ -60,6 +66,68 @@ export const defaultServices = {
 
   // Validation hooks for overriding
   async validateCustomSignUp(formData, touchData): Promise<ValidatorResult> {},
+  async validateFormPassword<Validator>(
+    formData,
+    touchData,
+    passwordSettings: PasswordSettings
+  ): Promise<ValidationComplexityError> {
+    const { password } = formData;
+
+    const { password: touched_password } = touchData;
+
+    /**
+     * If the password is not touched,
+     * or if the password settings are not set, we don't need to validate it.
+     */
+    if (!touched_password || !passwordSettings) return null;
+
+    const password_complexity = [];
+
+    const policyMinLength = +passwordSettings?.passwordPolicyMinLength;
+    if (password.length < policyMinLength) {
+      password_complexity.push(
+        `Password must contain at least ${policyMinLength} characters`
+      );
+    }
+
+    const passwordPolicyCharacters = passwordSettings?.passwordPolicyCharacters;
+
+    passwordPolicyCharacters?.forEach((errorCheck) => {
+      switch (errorCheck) {
+        case 'REQUIRES_LOWERCASE':
+          if (!/[a-z]/.test(password))
+            password_complexity.push(
+              `Password must contain a lower case letter`
+            );
+          break;
+        case 'REQUIRES_UPPERCASE':
+          if (!/[A-Z]/.test(password))
+            password_complexity.push(
+              `Password must contain an upper case letter`
+            );
+          break;
+        case 'REQUIRES_NUMBERS':
+          if (!/[0-9]/.test(password))
+            password_complexity.push(`Password must contain a number`);
+          break;
+        case 'REQUIRES_SYMBOLS':
+          if (!/[\W]/.test(password))
+            password_complexity.push(
+              `Password must contain a special character`
+            );
+          break;
+        default:
+          break;
+      }
+    });
+
+    /**
+     * Only return an error if there is at least one error.
+     */
+    return password_complexity.length !== 0
+      ? { password_complexity: password_complexity }
```

Comment:
Should this be `password` instead of `password_complexity`, since these are getting placed under the `password` field? IMO validation should map to each input name so that it matches one-to-one to each form input.

Reply:
OK! Yeah, I didn't quite understand how the naming worked, with the helper functions you already created. I'll change this then.

Reply:
Updated in 11fba8aca5ddf4546428a9d54b9f30d0cffa0ba8